use key codes

livibetter opened the pull request:
Self-explained [symbols](http://pubs.opengroup.org/onlinepubs/7908799/xcurses/curses.h.html#tag_000_000_258_008).

livibetter committed:
use key codes

ph1l reviewed (commented):
Looks sane... I'll test it out

ph1l commented on nstars.c:
```diff
@@ -157,7 +157,7 @@ int main(int argc, char *argv[])
 				}
 				break;
-			case 10:					/* Enter */
+			case KEY_ENTER:
```
This didn't work in my case. From ncurses getch manual page:

```
       KEY_ENTER  versus  control/M, KEY_BACKSPACE versus control/H.  Some curses
       implementations may differ according to whether they treat  these  control
       keys specially (and ignore the terminfo), or use the terminfo definitions.
       Ncurses uses the terminfo definition.  If it says that KEY_ENTER  is  con‐
       trol/M, getch will return KEY_ENTER when you press control/M.

       Generally, KEY_ENTER denotes the character(s) sent by the Enter key on the
       numeric keypad:

       ·   the terminal description lists the most useful keys,

       ·   the Enter key on the regular keyboard is already handled by the  stan‐
           dard ASCII characters for carriage-return and line-feed,

       ·   depending  on  whether  nl or nonl was called, pressing "Enter" on the
           regular keyboard may return either a carriage-return or line-feed, and
           finally

       ·   "Enter or send" is the standard description for this key.

```

So, it sound like the best way to handle all possibilities is:

```
diff --git a/nstars.c b/nstars.c
index 353ee4f..7037850 100644
--- a/nstars.c
+++ b/nstars.c
@@ -157,6 +157,8 @@ int main(int argc, char *argv[])
                                        beep();
                                }
                                break;
+                       case 10:                                        /* Line Feed */
+                       case 13:                                        /* Carriage Return */
                        case KEY_ENTER:
                                if (strncmp(cmd_buf_p, "/quit", MAX_CMD_BUF_LEN) == 0) {
                                        quit = 1;
```

livibetter replied:
Oh, terminals. I will add another commit, but I am thinking to use `'\n'` and `'\r'`[escapes][e] instead of the ASCII values, which probably would be moved into comments. What do you think?

[e]: https://en.wikipedia.org/wiki/Escape_sequences_in_C#Table_of_escape_sequences

 I am using Linux with st, I want to know what yours are, so if I stumble on this, I would know where to look into.

ph1l replied:
re: slash escapes -

static chars are ints, right? As long as they're equivalent, I think that's fine. You think that way will be more readable in the code?

re: platform

I run Debian Jessie and xterm. I'm not familiar with what "st" is..

livibetter replied:
[Yes][cint]. I do, I can never remember correctly which ASCII value belongs to which.

[cint]: http://stackoverflow.com/a/18422933/242583

[st][st] is "simple terminal" of suckless.

[st]: http://st.suckless.org/

livibetter committed:
handle ASCII Line Feed and Carriage Return

According to getch(3), the Enter key on regular keyboard returns the
ASCII \n and \r, the KEY_ENTER is intended for the Enter key on the
numeric keypad.

ph1l reviewed (approved):
Looks Good
